File: leetCode/0001 two_sum.js
// 1. Two Sum

var twoSum = function(nums, target) {
  let cache = {};
  for ( let i = 0 ; i < nums.length ; i++ ) {
    let c = target - nums[i];
    if ( c in cache ) return [cache[c], i];
    cache[nums[i]] = i;
  }
  return [];
};

File: leetCode/0001_two_sum.java
import java.util.HashMap;

class Solution {
  public int[] twoSum(int[] nums, int target) {
      var cache = new HashMap<Integer, Integer>();

      for ( int i = 0 ; i < nums.length ; i++ ) {
          var cur = target - nums[i];
          if ( cache.containsKey(cur) ) return new int[]{cache.get(cur), i};
          cache.put(nums[i], i);
      }

      return new int[]{};
  }
}